src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -2752,6 +2761,45 @@ static struct bin_attribute bin_attr_vmr_system_dtb = {
 	.size = 0
 };
 
+static ssize_t vmr_plm_sync_read(struct file *filp, struct kobject *kobj,
+	struct bin_attribute *attr, char *buf, loff_t off, size_t count)
+{
+	struct xocl_xgq_vmr *xgq =
+		dev_get_drvdata(container_of(kobj, struct device, kobj));
+	unsigned char *blob;
+	size_t size;
+	ssize_t ret = 0;
+
+	mutex_lock(&xgq->xgq_lock);
+	blob = xgq->xgq_vmr_plm_sync;
+	size = xgq->xgq_vmr_plm_sync_size;
+
+	if (off >= size)
+		goto out;
+
+	if (off + count > size)
+		count = size - off;
+	memcpy(buf, blob + off, count);
+	mutex_unlock(&xgq->xgq_lock);
```

Comment:
here, u will unlock twice. it will cause potential undefined behavior or make debugging harder. Please clean up your code.